yaki_mobile/lib/presentation/state/dio/dio_interceptor.dart

```diff
@@ -0,0 +1,37 @@
+import 'package:dio/dio.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+/// Interceptor are meant to be intercept request / response / error
+/// before they are handled by then or catch, and depending
+/// of the overwrited method, run custom action.
+/// Create a dio instance to add interceptor, which allow to use dynamic data
+final dioInterceptor = Provider(
+  (ref) {
+    final dio = Dio();
+    final addTokenToHeader = AddTokenToHeader();
+    dio.interceptors.add(
+      InterceptorsWrapper(
+        onRequest: (RequestOptions options, handler) =>
+            addTokenToHeader.onRequest(options, handler),
+      ),
+    );
+    return dio;
+  },
+);
```

Comment:
useless copy no ? if yes, remove it

Reply:
Its not a copy, its the dio interceptor, set it up it in its own file.